sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -476,7 +476,7 @@ case class StringRPad(str: Expression, len: Expression, pad: Expression)
 /**
  * Returns the input formatted according do printf-style format strings
  */
-case class StringFormat(children: Expression*) extends Expression with CodegenFallback {
+case class StringFormat(children: Expression*) extends Expression {
```

Comment:
No, I don't think so.  We just zip with `children` to check the types so I think it can work either way.  Just remember to subtract 1 when figuring out how many AnyDataTypes to fill in.